lldb/include/lldb/Target/SyntheticFrameProvider.h

```diff
@@ -0,0 +1,115 @@
+//===-- SyntheticFrameProvider.h --------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLDB_TARGET_SYNTHETICFRAMEPROVIDER_H
+#define LLDB_TARGET_SYNTHETICFRAMEPROVIDER_H
+
+#include "lldb/Core/PluginInterface.h"
+#include "lldb/Utility/Status.h"
+#include "lldb/lldb-forward.h"
+#include "llvm/Support/Error.h"
+
+#include <optional>
+#include <vector>
+
+namespace lldb_private {
+
+/// Descriptor for configuring a synthetic frame provider.
+///
+/// This struct contains the metadata needed to instantiate a frame provider
+/// and optional filters to control which threads it applies to.
+struct SyntheticFrameProviderDescriptor {
+  /// Metadata for instantiating the provider (e.g., script class name and args)
+  lldb::ScriptedMetadataSP scripted_metadata_sp;
+
+  /// Optional list of thread IDs to which this provider applies.
+  /// If empty, the provider applies to all threads.
+  std::vector<lldb::tid_t> thread_ids;
```

Comment:
You should allow thread names here as well.  Thread ID's aren't stable and so aren't going to be easy to use in general, but thread names can be known before running the binary and so are more useful for denoting "interesting threads".